system/Cache/Handlers/BaseHandler.php

```diff
@@ -14,12 +14,65 @@
 use Closure;
 use CodeIgniter\Cache\CacheInterface;
 use Exception;
+use InvalidArgumentException;
 
 /**
  * Base class for cache handling
  */
 abstract class BaseHandler implements CacheInterface
 {
+	/**
+	 * Reserved characters that cannot be used in a key or tag.
+	 *
+	 * @see https://github.com/symfony/cache-contracts/blob/c0446463729b89dd4fa62e9aeecc80287323615d/ItemInterface.php#L43
+	 */
+	public const RESERVED_CHARACTERS = '{}()/\@:';
+
+	/**
+	 * Maximum key length.
+	 */
+	public const MAX_KEY_LENGTH = PHP_INT_MAX;
+
+	/**
+	 * Prefix to apply to cache keys.
+	 * May not be used by all handlers.
+	 *
+	 * @var string
+	 */
+	protected $prefix;
+
+	/**
+	 * Validates a cache key according to PSR-6.
+	 * Keys that exceed MAX_KEY_LENGTH are hashed.
+	 *
+	 * @param string $key    The key to validate
+	 * @param string $prefix Optional prefix to include in length calculations
+	 *
+	 * @throws InvalidArgumentException When $key is not valid
+	 *
+	 * @see https://github.com/symfony/cache/blob/7b024c6726af21fd4984ac8d1eae2b9f3d90de88/CacheItem.php#L158
+	 */
+	public static function validateKey($key, $prefix = ''): string
+	{
+		if (! is_string($key))
+		{
+			throw new InvalidArgumentException('Cache key must be a string');
+		}
+		if ($key === '')
+		{
+			throw new InvalidArgumentException('Cache key cannot be empty.');
+		}
+		if (strpbrk($key, self::RESERVED_CHARACTERS) !== false)
+		{
+			throw new InvalidArgumentException('Cache key contains reserved characters ' . self::RESERVED_CHARACTERS);
+		}
+
+		// If the key with prefix exceeds the length then return the hashed version
+		return strlen($prefix . $key) > static::MAX_KEY_LENGTH ? $prefix . md5($key) : $prefix . $key;
```

Comment:
Since we're checking against the length of `$prefix . $key`, should the hash include the prefix also?
```suggestion
		return strlen($prefix . $key) > static::MAX_KEY_LENGTH ? md5($prefix . $key) : $prefix . $key;
```

Reply:
No, the prefix may be crucial to some handlers for the end result (i.e. an optimized `FileHandler` may organize folder structures using the prefix as a starting point) so we need to be sure it ends up as the beginning of the final key. It is included in the length calculation to mitigate accidental overages: e.g. `FileHandler` has a 255 limit because of the filesystem, so if you pass a 250 key with a 10 prefix it needs to be hashed.